cmake/PythonUtils.cmake

```diff
@@ -0,0 +1,86 @@
+# TODO: Eventually, we want to automatically turn on this auto-pip install functionality. That
+# should only be done when we KNOW that we are in an isolated configuration of Python.
+option(ENABLE_PIP_INSTALL "Allow automatic installation of Python dependency modules." OFF)
+
+function(python_test_modules)
+    cmake_parse_arguments(_ptm "REQUIRED" "" "MODULES" ${ARGN})
+
+    if(NOT _ptm_MODULES)
+        message(FATAL_ERROR "python_test_modules() must be given a list of MODULES")
+    endif()
+
+    foreach(_module_NAME IN LISTS _ptm_MODULES)
+        if(Python3_EXECUTABLE)
+            execute_process(
+                COMMAND "${Python3_EXECUTABLE}" -c "import ${_module_NAME}"
+                RESULT_VARIABLE RETURNCODE
+                OUTPUT_QUIET ERROR_QUIET
+            )
+            if(NOT RETURNCODE)
+                set(${_module_NAME}_FOUND ON PARENT_SCOPE)
+            else()
+                set(${_module_NAME}_FOUND OFF PARENT_SCOPE)
+            endif()
+        else()
+            set(${_module_NAME}_FOUND OFF PARENT_SCOPE)
+        endif()
+
+        if(_ptm_REQUIRED AND NOT ${_module_NAME}_FOUND)
+            message(FATAL_ERROR "Could NOT find Python3 Module: ${_module_NAME}")
+        endif()
+    endforeach()
+endfunction()
+
+function(python_install_modules)
+    cmake_parse_arguments(_pim "" "REQUIREMENTS_FILE" "MODULES" ${ARGN})
+
+    if(NOT _pim_REQUIREMENTS_FILE AND NOT _pim_MODULES)
+        message(FATAL_ERROR "python_install_modules() requires either a REQUIREMENTS_FILE or a list of MODULES")
+    endif()
+    if(NOT Python3_EXECUTABLE)
+        message(WARNING "Python3 is not available, not installing requested modules.")
+    endif()
+
+    # Escape hatch for folks who don't want us mucking around with their system Python.
+    if(NOT ENABLE_PIP_INSTALL)
```

Comment:
I wonder if this should be moved to a higher location (e.g. around the call to `python_install_modules`, rather than within it)...  Like Deledrius, I was surprised to see an unprotected global pip install, since this line wasn't obvious at first glance.

Reply:
I was thinking that I wanted this to be future proof in case in that the next person who uses this function wouldn't need to guard it. Maybe, as a compromise, this could be merged with python_test_modules such that modules could be installed from a requirements file before being tested?